src/graphql/mutations/CreateArticleCategory.js

```diff
@@ -38,10 +48,50 @@ export async function createArticleCategory({
     updatedAt: now,
   };
 
-  // TODO: Insert articleCategory to articles in Elasticsearch, and return insertion results from DB.
-  // Implementation can refer to createArticleReply() in CreateArticleReply.js
-  //
-  return (article.articleCategories || []).concat(articleCategory); // MOCK
+  const {
+    result: articleResult,
+    get: { _source },
+  } = await client.update({
+    index: 'articles',
+    type: 'doc',
+    id: article.id,
+    body: {
+      script: {
+        /**
+         * Check if the category is already connected in the article.
+         * If so, do nothing;
+         * otherwise, do update.
+         */
+        source: `
+          if(
+            ctx._source.articleCategories.stream().anyMatch(
+              ar -> ar.get('categoryId').equals(params.articleCategory.get('categoryId'))
+            )
+          ) {
+            ctx.op = 'none';
```

Comment:
What if the matched `articleCategories` is a deleted one? according to our [previous discussion](https://github.com/cofacts/rumors-api/issues/143#issuecomment-592921145), its user should be updated and its status should be back to `NORMAL`.

Reply:
To see
38bd966a7aa3dfb3dbe06ddeebba2ff7da11da1f
782fe61fc476c2fc71921139e940d20a943193c2